server/src/main/java/org/opensearch/cluster/routing/remote/RemoteRoutingTableService.java

```diff
@@ -0,0 +1,104 @@
+ *
+ * @opensearch.internal
+ */
+public class RemoteRoutingTableService implements Closeable {
+
+    /**
+     * Cluster setting to specify if routing table should be published to remote store
+     */
+    public static final Setting<Boolean> REMOTE_ROUTING_TABLE_ENABLED_SETTING = Setting.boolSetting(
+        "cluster.remote_store.routing.enabled",
+        false,
+        Setting.Property.NodeScope,
+        Setting.Property.Final
+    );
+    private static final Logger logger = LogManager.getLogger(RemoteRoutingTableService.class);
+    private final Settings settings;
+    private final Supplier<RepositoriesService> repositoriesService;
+    private final ClusterSettings clusterSettings;
+    private BlobStoreRepository blobStoreRepository;
+
+    public RemoteRoutingTableService(Supplier<RepositoriesService> repositoriesService,
+                                     Settings settings,
+                                     ClusterSettings clusterSettings) {
+        assert isRemoteRoutingTableEnabled(settings) : "Remote routing table is not enabled";
+        this.repositoriesService = repositoriesService;
+        this.settings = settings;
+        this.clusterSettings = clusterSettings;
+    }
+
+    public List<ClusterMetadataManifest.UploadedIndexMetadata> writeFullRoutingTable(ClusterState clusterState, String previousClusterUUID) {
+        return null;
+    }
+
+    public List<ClusterMetadataManifest.UploadedIndexMetadata> writeIncrementalMetadata(
+        ClusterState previousClusterState,
+        ClusterState clusterState,
+        ClusterMetadataManifest previousManifest) {
+        return null;
+    }
+
+    public RoutingTable getLatestRoutingTable(String clusterName, String clusterUUID) {
+        return null;
+    }
+
+    public RoutingTable getIncrementalRoutingTable(ClusterState previousClusterState, ClusterMetadataManifest previousManifest, String clusterName, String clusterUUID) {
+        return null;
+    }
+
+    private void deleteStaleRoutingTable(String clusterName, String clusterUUID, int manifestsToRetain) {
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (blobStoreRepository != null) {
+            IOUtils.close(blobStoreRepository);
+        }
+    }
+
+    public void start() {
+        assert isRemoteRoutingTableEnabled(settings) == true : "Remote routing table is not enabled";
+        final String remoteStoreRepo = settings.get(
+            Node.NODE_ATTRIBUTES.getKey() + RemoteStoreNodeAttribute.REMOTE_STORE_ROUTING_TABLE_REPOSITORY_NAME_ATTRIBUTE_KEY
+        );
+        assert remoteStoreRepo != null : "Remote routing table repository is not configured";
+        final Repository repository = repositoriesService.get().repository(remoteStoreRepo);
+        assert repository instanceof BlobStoreRepository : "Repository should be instance of BlobStoreRepository";
+        blobStoreRepository = (BlobStoreRepository) repository;
+    }
```

Comment:
What is in this to start? Can this be a part of ctor itself?

Reply:
We are initializing the `blobStoreRepo` here, we can do it during init as well, but following how it is done in `RemoteClusterStateService` to keep it consistent.